rust/tool/src/generation.rs

```diff
@@ -40,10 +40,28 @@ pub struct Generation {
 impl Generation {
     pub fn from_link(link: &GenerationLink) -> Result<Self> {
         let bootspec_path = link.path.join("boot.json");
-        let generation: BootspecGeneration = serde_json::from_slice(
-            &fs::read(bootspec_path).context("Failed to read bootspec file")?,
-        )
-        .context("Failed to parse bootspec json")?;
+        let generation: BootspecGeneration;
```

Comment:
Just a nit as a passer-by: you can prevent having this "uninitialized" variable by binding it to the `if let Ok(...)` below:

```rust
let generation = if let Ok(...) {
  serde_json(....)
} else {
  BootspecGeneration::V1(...)
};
```

Reply:
It's not really "uninitialized" because Rust does not allow uninitialized variables, it's equivalent to what you wrote, I just find it cleaner to not have the `=` thing, hmm I don't have strong opinion on that

Reply:
It doesn't initially have a value, so it's uninitialized, to me :P

I think the `let ... = if let` pattern is more idiomatic Rust, but it doesn't functionally change anything, so you do you!

Reply:
(irrelevant due to code changes)